OptimizelySDK.Tests/DecisionServiceTest.cs

```diff
@@ -441,5 +452,131 @@ public void TestGetVariationWithBucketingId()
             actualVariationKey = optlyObject.GetVariation(experimentKey, userId, userAttributesWithBucketingId);
             Assert.AreEqual(variationKeyControl, actualVariationKey, string.Format("Variation \"{0}\" does not match expected user profile variation \"{1}\".", actualVariationKey, variationKeyControl));
         }
+
+
+        #region GetVariationForFeature() method Tests
+
+        // Should return null and log a message when the feature flag's experiment ids array is empty
+        [Test]
+        public void TestGetVariationForFeatureExperimentGivenNullExperimentIds()
+        {
+            var featureFlag = ProjectConfig.GetFeatureFlagFromKey("empty_feature");
+
+            Assert.IsNull(DecisionService.GetVariationForFeatureExperiment(featureFlag, GenericUserId, new UserAttributes() { }));
+
+            LoggerMock.Verify(l => l.Log(LogLevel.INFO, $"The feature flag \"{featureFlag.Key}\" is not used in any experiments."));
+        }
+
+        // Should return null and log when the experiment is not in the datafile 
+        [Test]
+        public void TestGetVariationForFeatureExperimentGivenExperimentNotInDataFile()
+        {
+            var featureFlagWithInvalidExperiment = new FeatureFlag { Id = "29039203", Key = "boolean_feature", RolloutId = "abc", ExperimentIds = new List<string> { "29039203" } };
```

Comment:
please don't use like this. modify JSON & then get feature flag from the config file.

Reply:
Done